File: admin/app/routers/scare.py
import json
from fastapi import APIRouter, Request

router = APIRouter()


@router.get("/api/scare/{zone}/config")
def get_scare_config(zone: str, request: Request):
    row = request.app.state.db.execute(
        "SELECT enabled_hashes FROM scare_zone_config WHERE zone = ?", (zone,)
    ).fetchone()
    if row is None:
        return {"enabled_hashes": []}
    return {"enabled_hashes": json.loads(row[0])}


@router.put("/api/scare/{zone}/config")
async def put_scare_config(zone: str, request: Request):
    body = await request.json()
    enabled_hashes = body.get("enabled_hashes", [])
    db = request.app.state.db
    db.execute(
        """INSERT INTO scare_zone_config (zone, enabled_hashes) VALUES (?, ?)
           ON CONFLICT(zone) DO UPDATE SET enabled_hashes=excluded.enabled_hashes""",
        (zone, json.dumps(enabled_hashes)),
    )
    db.commit()
    request.app.state.bridge.publish_scare_config(zone, enabled_hashes)
    return {"ok": True}


@router.post("/api/scare/{zone}/test")
def post_scare_test(zone: str, request: Request):
    request.app.state.bridge.publish_scare_test(zone)
    return {"ok": True}

File: admin/frontend/src/api/scare.ts
import { apiFetch } from "./client";
import type { ScareConfig } from "../types";

export function getScareConfig(zone: string): Promise<ScareConfig> {
  return apiFetch<ScareConfig>(`/api/scare/${encodeURIComponent(zone)}/config`);
}

export function putScareConfig(zone: string, config: ScareConfig): Promise<void> {
  return apiFetch(`/api/scare/${encodeURIComponent(zone)}/config`, {
    method: "PUT",
    body: JSON.stringify(config),
  });
}

export function testScare(zone: string): Promise<void> {
  return apiFetch(`/api/scare/${encodeURIComponent(zone)}/test`, { method: "POST" });
}
